addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -318,18 +319,36 @@ async function loadImageInfo(img, rpc) {
     // Check that url is local.
     const isLocal = original && new URL(original.image_src, window.location.origin).origin === window.location.origin;
     if (isLocal && original.image_src) {
+        const canAddShapeOnImage = !!img.dataset.shape && isImageMimetypeSupported(original.mimetype);
         img.dataset.originalId = original.id;
         img.dataset.originalSrc = original.image_src;
-        img.dataset.mimetype = original.mimetype;
+        img.dataset.mimetype = canAddShapeOnImage ? 'image/svg+xml' : original.mimetype;
+        if (!canAddShapeOnImage) {
+            delete img.dataset.shape;
+            delete img.dataset.shapeColors;
+            delete img.dataset.fileName;
+            delete img.dataset.originalMimetype;
+            return;
+        }
+        img.dataset.originalMimetype = original.mimetype;
     }
 }
 
+/**
+ * @param {String} mimetype
+ * @returns {Boolean}
+ */
+function isImageMimetypeSupported(mimetype) {
+    return supportedImageMimetypes.includes(mimetype);
+}
```

Comment:
Seems kinda random name for this generic function... supported to do what? All image processing stuff?